integration-tests/src/test/java/org/wildfly/prospero/extended/MultiUserServerDirectoryTest.java

```diff
@@ -0,0 +1,239 @@
+                .withCopyFileToContainer(MountableFile.forHostPath(updatedManifest.toPath()), "/dist/manifest.yaml")
+                .withStartupTimeout(Duration.ofSeconds(120));
+
+        container.start();
+        filesystem = new ContainerFilesystem(container);
+
+        // prepare the container by creating useraccounts and directory structure
+        serveradmin = new User("serveradmin", container);
+        siteadmin = new User("siteadmin", container);
+
+        container.execInContainer("cp", "-r", "/dist/wildfly", "/home/serveradmin/");
+        container.execInContainer("cp", "-r", "/dist/prospero", "/home/serveradmin/");
+        container.execInContainer("cp", "-r", "/dist/repository", "/home/serveradmin/");
+    }
+
+    static boolean isDockerAvailable() {
+        try {
+            DockerClientFactory.instance().client();
+            return true;
+        } catch (Throwable ex) {
+            return false;
+        }
+    }
+
+    @After
+    public void tearDown() throws IOException, InterruptedException {
+        if (filesystem != null && filesystem.fileExists("/home/serveradmin/prospero/logs/installation.log")) {
+            container.copyFileFromContainer("/home/serveradmin/prospero/logs/installation.log", "target/installation.log");
+        }
+        if (container != null) {
+            container.stop();
+        }
+    }
+
+    @Test
+    public void updateRollbackDoesNotRemoveExternalFiles() throws Exception {
+        // actual test setup - make the server and prospero owned by 'serveradmin'
+        container.execInContainer("chmod", "0777", "serveradmin:serveradmin", "/home/serveradmin");
+        container.execInContainer("chmod", "0777", "serveradmin:serveradmin", "/home/serveradmin/wildfly");
+        container.execInContainer("chown", "-R", "serveradmin:serveradmin", "/home/serveradmin");
+
+        // actual test setup - create the file owned be a different user
+        siteadmin.execute("touch", "/home/serveradmin/wildfly/test");
+        siteadmin.execute("chmod", "0755", "/home/serveradmin/wildfly/test");
+
+        // make the apply fail on purpose
+        container.execInContainer("chmod", "-R", "0555", "serveradmin:serveradmin", "/home/serveradmin/wildfly/modules");
+
+        // finally perform update
+        final Container.ExecResult updateRes = serveradmin.tryExecute("/home/serveradmin/prospero/bin/prospero.sh", "update", "perform",
+                "--dir", "/home/serveradmin/wildfly", "--repositories", "/home/serveradmin/repository/", "-y", "-vv");
+        assertEquals(updateRes.toString(), 1, updateRes.getExitCode());
+
+        final Container.ExecResult res = container.execInContainer("ls", "-alF", "/home/serveradmin/wildfly");
+        assertTrue(res.toString(), filesystem.fileExists("/home/serveradmin/wildfly/test"));
+    }
+
+    @Test
+    public void updateServerWithAnExternalFolderNotOwnedByUser() throws Exception {
```

Comment:
What is the "external folder"? The "/home/serveradmin/wildfly/test" looks to be a file.

Reply:
Good point, I'll add a folder owned by the siteadmin to cover that case.